src/PAModel/CanvasDocument.cs

```diff
@@ -499,6 +537,12 @@ private void RestoreAssetFilePaths()
                     msappFileName = maxFileNumber.ToString("D4") + assetFilePath.GetExtension();
                 }
 
+                if (resource.OriginalName != null)
+                {
+                    resource.Name = resource.OriginalName;
+                }
+
+                resource.OriginalName = null;
                 var updatedPath = FilePath.FromMsAppPath(Utilities.GetResourceRelativePath(resource.Content)).Append(msappFileName);
                 resource.Path = updatedPath.ToMsAppPath();
                 resource.FileName = msappFileName;
```

Comment:
Are the mutations here safe? 
We get here via: SaveAsMsapp() --> ApplyBeforeMsAppWriteTransforms() --> RestoreAssetFilePaths(). 

Ie, does this cause a mutation of the app._resource after we call save?

Reply:
basically yes it's safe it's guarded the same way the other transforms are